src/rp2_common/hardware_uart/include/hardware/uart.h

```diff
@@ -310,7 +310,8 @@ static inline bool uart_is_readable(uart_inst_t *uart) {
 /*! \brief  Write to the UART for transmission.
  *  \ingroup hardware_uart
  *
- * This function will block until all the data has been sent to the UART
+ * This function will block until all the data has been written to the UART.
+ * To check for UART Tx FIFO empty, please see: uart_tx_wait_blocking()
```

Comment:
Might be worth (explicitly) noting that the UART _may_ still be transmitting (i.e. Tx FIFO not empty) even after this function has returned?
Perhaps if the first line changed "UART" to "UART block" that might help differentiate between UART as the peripheral-block, and UART as the physical-wires ? :man_shrugging: